ref(profiling): forward profiles of non-sampled transactions (with no options filtering)

viglia opened the pull request:
Remove the options to shutdown, filter by platform or downsample the number of processed profiles we pass along to extract the metrics.

With this, all the `Processed Profiles` (that are not indexed), will be passed along, as we currently do, but without the further checks that are currently in place (*rollout rate*, *platforms*, etc.).

The global kill-switch is left, should we ever need it in case of incidents.

Commit:
ingest all unsampled profiles

remove the options to shutdown, filter by platform or downsample the
number of processed profiles we pass along to extract the metrics.

Commit:
remove unused GlobalConfig

Commit:
initialize global_gonfig in if_processing!

Commit:
remove indexedProfile filtered outcome from test

viglia commented on tests/integration/test_outcome.py:
```diff
@@ -1277,16 +1277,6 @@ def make_envelope(transaction_name):
-            "quantity": 1,
-            "reason": "Sampled:3000",
-            "source": expected_source,
-        },
```
@jjbayer @Dav1dde please, can you double-check if the removal of this is correct as I'm not too familiar with this integration test.

Dav1dde replied:
Yes this is correct if the profile is now forwarded to storage! (A sentry consumer would then create the accepted outcome).

viglia replied:
@Dav1dde I don't want to create confusion. For clarity: this PR is supposed to leave things as they are, in the sense that `processed` profiles are still meant to be passed along all the way through the pipeline only for metrics extraction. We're not going to store the profile itself in `GCS`.

The Goal here is just to remove any *FeatureFlag* or *Options* as we're now sure we want to get all the processed ones (no filters needed: sampling, platform, etc.).

jjbayer replied:
@viglia how did you manage to make this diff disappear?

viglia replied:
re-introduced the global kill switch, so the test made sense how it was before (if it's disabled it should be filtered out), hence I re-introduced this outcome too.

jjbayer commented on relay-server/src/services/processor.rs:
```diff
@@ -1776,18 +1774,14 @@ impl EnvelopeProcessorService {
             self.extract_transaction_metrics(state, SamplingDecision::Drop, profile_id)?;
 
-            dynamic_sampling::drop_unsampled_items(state, outcome, keep_profiles);
+            dynamic_sampling::drop_unsampled_items(state, outcome, true);
```
If we go through with this change, let's also remove the boolean from the function signature.

jjbayer commented on relay-server/src/services/processor/profile.rs:
```diff
@@ -19,7 +19,7 @@ use crate::utils::ItemAction;
     let profiling_disabled = state.should_filter(Feature::Profiling);
     let has_transaction = state.event_type() == Some(EventType::Transaction);
-    let keep_unsampled_profiles = !state.should_filter(Feature::IngestUnsampledProfiles);
+    let keep_unsampled_profiles = true;
```
See comment above.

jjbayer reviewed (commented):
I like this change because it reduces complexity. But are we confident that we can sustain ingesting all profiles for the foreseeable future? Would it make sense to keep at least the global kill switch so we have some control during incidents?

Commit:
add back global kill switch

add back `profiling.profile_metrics.unsampled_profiles.enabled` global
kill switch.

Commit:
fix test

Commit:
update changelog

Commit:
Merge remote-tracking branch 'origin' into viglia/ref/remove-options-and-ingest-all-unsampled-profiles

jjbayer commented on relay-dynamic-config/src/feature.rs:
```diff
@@ -61,13 +61,6 @@ pub enum Feature {
-    ///
-    /// Serialized as `projects:profiling-ingest-unsampled-profiles`.
-    #[serde(rename = "projects:profiling-ingest-unsampled-profiles")]
-    IngestUnsampledProfiles,
```
Something I forgot to point out on first review: We should keep sending this flag at least until the next self-hosted release, because external relays will not have the new code yet and will stop sending unsampled profiles entirely. See list of `Deprecated*` features at the bottom of this list.

Same goes for the global options unfortunately.

jjbayer commented on relay-ua/uap-core:
Please make sure `uap-core` is on the same commit as on the master branch so this PR does not accidentally revert the version bump.

Commit:
define deprecated feature

add deprecated feature for projects:profiling-ingest-unsampled-profiles

Commit:
add deprecated global options

define the following as deprecated:
* `profiling.profile_metrics.unsampled_profiles.platforms`
* `profiling.profile_metrics.unsampled_profiles.sample_rate`

Commit:
update relay-ua/uap-core submodule to v0.18.0

jjbayer reviewed (approved):
Please fix the `uap-core` diff as discussed, after that this looks good to go!

Commit:
Merge remote-tracking branch 'origin' into viglia/ref/remove-options-and-ingest-all-unsampled-profiles
